src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3WebtransportManager.cs

```diff
@@ -0,0 +1,120 @@
+        {
+            return _sessions.TryGetValue(streamId, out session);
+        }
+
+        public void AcceptServerStream(QuicStream stream, long sessionId)
+        {
+            Http3WebtransportSession? session;
+            _sessions.TryGetValue(sessionId, out session);
+            // if no session with that id exists throw exception
+            // https://datatracker.ietf.org/doc/html/draft-ietf-webtrans-http3#section-4
+            if (session == null)
+            {
+                stream.Abort(QuicAbortDirection.Both, (long)Http3ErrorCode.WebtransportBufferedStreamRejected);
+                return;
+            }
+            session.AcceptServerStream(stream);
+        }
+
+        public async ValueTask<QuicStream?> CreateClientStream(QuicStreamType type, long sessionId)
+        {
+            QuicStream clientWebtransportStream;
+            try
+            {
+                clientWebtransportStream = await _connection.OpenOutboundStreamAsync(type).ConfigureAwait(false);
+                await clientWebtransportStream.WriteAsync(BuildWebtransportStreamClientFrame(type, sessionId), default).ConfigureAwait(false);
+
+                return clientWebtransportStream;
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+        }
+
+        private byte[] BuildWebtransportStreamClientFrame(QuicStreamType type, long sessionId)
+        {
+            Span<byte> buffer = stackalloc byte[2 + VariableLengthIntegerHelper.MaximumEncodedLength];
+            int webtransportLength;
+            if (type == QuicStreamType.Unidirectional)
+                webtransportLength = VariableLengthIntegerHelper.WriteInteger(buffer.Slice(0), (long)Http3StreamType.WebTransportUnidirectional);
+            else
+                webtransportLength = VariableLengthIntegerHelper.WriteInteger(buffer.Slice(0), (long)Http3StreamType.WebTransportBidirectional);
+            int webtransportSessionLength = VariableLengthIntegerHelper.WriteInteger(buffer.Slice(2), (long)sessionId);
+            int payloadLength = webtransportLength + webtransportSessionLength; // includes the webtransport stream and the session id
+            Debug.Assert(payloadLength <= VariableLengthIntegerHelper.OneByteLimit);
+
+            return buffer.Slice(0, payloadLength).ToArray();
+        }
+
+        public void DeleteSession(long id)
+        {
+            _sessions.TryRemove(id, out _);
+        }
+
+        public async ValueTask DisposeAsync()
+        {
+            List<long> toRemove = new List<long>();
+            foreach (KeyValuePair<long, Http3WebtransportSession> pair in _sessions)
+            {
+               toRemove.Add(pair.Key);
+            }
+
+            foreach (long id in toRemove)
+            {
+                await _sessions[id].DisposeAsync().ConfigureAwait(false);
+            }
+        }
+        public void Dispose()
```

Comment:
What do you think should happen if Dispose is called twice?

And more interestingly:
What do you think should happen if you call AddSession after Dispose?
What do you think should happen if Dispose is called while you are in the middle of AddSession call?

Reply:
I thought that disposing the manager only once when we dispose the connection would avoid a calling of Dispose twice, 
But I think if Dispose is called twice at the same time then some sessions from the dictionary would happen to be disposed twice.
I modified the code to clear the dictionary after putting the sessions in the list

Having the fact that AddSession is called inside sendAsync I thought that the manager cannot be disposed unless the Quic  connection is being closed and that would close my request stream anyway, but yes that could still mess with my dictionary
But I think a lock on the dictionary (when adding a sesion and on disposal) and a `private _disposed` variable  would solve the cases.

Reply:
@CarnaViire implemented